packages/cli/src/server/api/memory/groups.ts

```diff
@@ -0,0 +1,134 @@
+import type { IAgentRuntime, UUID } from '@elizaos/core';
+import { validateUuid, logger, createUniqueUuid, ChannelType } from '@elizaos/core';
+import express from 'express';
+import type { AgentServer } from '../../index';
+import { sendError, getRuntime } from '../shared';
+
+/**
+ * Group and world memory management functionality
+ */
+export function createGroupMemoryRouter(
+  agents: Map<UUID, IAgentRuntime>,
+  serverInstance: AgentServer
+): express.Router {
+  const router = express.Router();
+  const db = serverInstance?.database;
+
+  // Create group memory spaces for multiple agents
+  router.post('/groups/:serverId', async (req, res) => {
+    const serverId = validateUuid(req.params.serverId);
+    const { name, worldId, source, metadata, agentIds = [] } = req.body;
+
+    if (!Array.isArray(agentIds) || agentIds.length === 0) {
+      return sendError(res, 400, 'BAD_REQUEST', 'agentIds must be a non-empty array');
+    }
+
+    let results = [];
+    let errors = [];
+
+    for (const agentId of agentIds) {
+      try {
+        const runtime = getRuntime(agents, agentId as UUID);
+        const roomId = createUniqueUuid(runtime, serverId as string);
+        const roomName = name || `Chat ${new Date().toLocaleString()}`;
+
+        await runtime.ensureWorldExists({
+          id: worldId,
+          name: source,
+          agentId: runtime.agentId,
+          serverId: serverId as UUID,
+        });
+
+        await runtime.ensureRoomExists({
+          id: roomId,
+          name: roomName,
+          source,
+          type: ChannelType.API,
+          worldId,
+          serverId: serverId as UUID,
+          metadata,
+          channelId: roomId,
+        });
+
+        await runtime.addParticipant(runtime.agentId, roomId);
+        await runtime.ensureParticipantInRoom(runtime.agentId, roomId);
+        await runtime.setParticipantUserState(roomId, runtime.agentId, 'FOLLOWED');
```

Comment:
nit: I see this many places, seems could consolidate to just addParticipant, ensuring that is inside room after is bit bloat, if added participant, they should be inside a room with await.